projects/vdk-plugins/vdk-structlog/README.md

```diff
@@ -23,11 +23,12 @@ pip install vdk-structlog
 | structlog_metadata            | Configure the metadata that will be output along with the log message                                                                                                                                                            | "timestamp, level, logger_name, file_name, vdk_job_name | Any combination of the following: "timestamp, level, logger_name, file_name, line_number, function_name, vdk_job_name, vdk_step_name, vdk_step_type". Can be expanded by extra params and bound key-value pairs. See the bound logger examples for more information |
 | structlog_format              | Configure the logging output format. Available formats: json, console, ltsv                                                                                                                                                      | "console"                                               | "console", "json", "ltsv"                                                                                                                                                                                                                                           |
 | structlog_console_log_pattern | Custom format string for console logging, applied only when`logging_format` is 'console'. Overrides `logging_metadata`. Note: For config.ini, %-signs should be escaped by doubling, e.g. %(asctime)s should become %%(asctime)s | "%(asctime)s %(name)-12s %(levelname)-8s %(message)s"   | Any valid Python logging format string                                                                                                                                                                                                                              |
 | structlog_config_preset       | Choose a configuration preset. Any config options set together with the preset will override the preset options. Available presets: LOCAL, CLOUD.                                                                                | "CLOUD"                                                 | "console", "json", "ltsv"                                                                                                                                                                                                                                           |
+| structlog_format_init_logs    | Set to True to apply structlog formatting options to vdk initialization logs                                                                                                                                                     | "True"                                                  | "True", "False"                                                                                                                                                                                                                                                     |
```

Comment:
But we want to make sure all logs are formatted the same. 

I mean if I do `vdk list` or `vdk deploy` even if deploy or list logs something after vdk initialization we still would like them to be the same (without extra formatting)

It's quite possible that I am misunderstanding the name of the option and the description. It seems to me to imply that this option control how vdk initialization logs (logs during vdk initiazation are formatted) ?

Reply:
Data job plugins have the 5 basic hooks

```
vdk_configure
vdk_initialize
initialize_job
run_job
run_step
```

If we configure logging with all the structlog options at `vdk_initialize`, then anything between vdk_initialize and intialize_job will also be formatted (which is the command output you pointed to in the ticket). So we should configure logging at `initialize_job` instead, which is what vdk_core does. It will affect just the job-related logs, which is what we're looking for, right?

https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/config/log_config.py#L292

Reply:
Of course, the result for job runs is not ideal.

Configure logging on `vdk_initialize`

```
Versatile Data Kit (VDK)
Version: 0.3.1166422902
Build details: RELEASE_VERSION=0.3.1166422902, BUILD_DATE=Tue Feb  6 15:11:46 UTC 2024, BUILD_MACHINE_INFO=Linux runner-trzkizxz-project-28359933-concurrent-0dlv2t 5.4.235-144.344.amzn2.x86_64 #1 SMP Sun Mar 12 12:50:22 UTC 2023 x86_64 GNU/Linux, GITLAB_CI_JOB_ID=6107678693, GIT_COMMIT_SHA=b3e7b45b38eaec7db001baaa38b9ff0b9ad30366, GIT_BRANCH=main
Python version: 3.11.7 64bit (/Users/mdilyan/Projects/versatile-data-kit/projects/vdk-plugins/vdk-structlog/venv/bin/python)

Installed plugins:
vdk-ingest-http (from package vdk-ingest-http, version 0.2.1156222304)
vdk-sqlite (from package vdk-sqlite, version 0.1.1156222304)
vdk-server (from package vdk-server, version 0.1.948436673)
vdk-structlog (from package vdk-structlog, version 0.1.0)
vdk-control-service-properties (from package vdk-plugin-control-cli, version 0.1.948436673)
vdk-execution-skip (from package vdk-plugin-control-cli, version 0.1.948436673)
vdk-plugin-control-cli (from package vdk-plugin-control-cli, version 0.1.948436673)
vdk-ingest-file (from package vdk-ingest-file, version 0.1.948436673)
--------------------------------------------------------------------------------
Run job with directory /Users/mdilyan/Projects/vdk-playground/hello-world
{"name": "vdk.plugin.control_cli_plugin.properties_plugin", "filename": "properties_plugin.py", "lineno": 30, "funcName": "initialize_job", "message": "Control Service REST API URL is not configured. Will not initialize Control Service based Properties client implementation.", "level": "INFO", "timestamp": 1707826485.856691}
{"name": "vdk.plugin.control_cli_plugin.execution_skip", "filename": "execution_skip.py", "lineno": 105, "funcName": "_skip_job_if_necessary", "message": "Checking if job should be skipped:", "level": "INFO", "timestamp": 1707826485.856889}
{"name": "vdk.plugin.control_cli_plugin.execution_skip", "filename": "execution_skip.py", "lineno": 106, "funcName": "_skip_job_if_necessary", "message": "Job : hello-world, Team : my-team, Log config: LOCAL, execution_id: 6fdb3756-2455-429b-a525-fa869fdb9545-1707826485", "level": "INFO", "timestamp": 1707826485.8569238}
```

Configure logging on initialize_job
```
{"name": "vdk.internal.cli_entry", "filename": "cli_entry.py", "lineno": 135, "funcName": "vdk_main", "message": "Start CLI vdk with args ['run', 'hello-world']", "level": "INFO", "timestamp": 1707826531.086813}
{"name": "vdk.internal.builtin_plugins.run.cli_run", "filename": "cli_run.py", "lineno": 269, "funcName": "run", "message": "Versatile Data Kit (VDK)\nVersion: 0.3.1166422902\nBuild details: RELEASE_VERSION=0.3.1166422902, BUILD_DATE=Tue Feb  6 15:11:46 UTC 2024, BUILD_MACHINE_INFO=Linux runner-trzkizxz-project-28359933-concurrent-0dlv2t 5.4.235-144.344.amzn2.x86_64 #1 SMP Sun Mar 12 12:50:22 UTC 2023 x86_64 GNU/Linux, GITLAB_CI_JOB_ID=6107678693, GIT_COMMIT_SHA=b3e7b45b38eaec7db001baaa38b9ff0b9ad30366, GIT_BRANCH=main\nPython version: 3.11.7 64bit (/Users/mdilyan/Projects/versatile-data-kit/projects/vdk-plugins/vdk-structlog/venv/bin/python)\n\nInstalled plugins:\nvdk-ingest-http (from package vdk-ingest-http, version 0.2.1156222304)\nvdk-sqlite (from package vdk-sqlite, version 0.1.1156222304)\nvdk-server (from package vdk-server, version 0.1.948436673)\nvdk-structlog (from package vdk-structlog, version 0.1.0)\nvdk-control-service-properties (from package vdk-plugin-control-cli, version 0.1.948436673)\nvdk-execution-skip (from package vdk-plugin-control-cli, version 0.1.948436673)\nvdk-plugin-control-cli (from package vdk-plugin-control-cli, version 0.1.948436673)\nvdk-ingest-file (from package vdk-ingest-file, version 0.1.948436673)\n--------------------------------------------------------------------------------", "level": "INFO", "timestamp": 1707826531.108256}
{"name": "vdk.internal.builtin_plugins.run.cli_run", "filename": "cli_run.py", "lineno": 193, "funcName": "create_and_run_data_job", "message": "Run job with directory /Users/mdilyan/Projects/vdk-playground/hello-world", "level": "INFO", "timestamp": 1707826531.1085}
{"name": "vdk.plugin.control_cli_plugin.properties_plugin", "filename": "properties_plugin.py", "lineno": 30, "funcName": "initialize_job", "message": "Control Service REST API URL is not configured. Will not initialize Control Service based Properties client implementation.", "level": "INFO", "timestamp": 1707826531.111215}
{"name": "vdk.plugin.control_cli_plugin.execution_skip", "filename": "execution_skip.py", "lineno": 105, "funcName": "_skip_job_if_necessary", "message": "Checking if job should be skipped:", "level": "INFO", "timestamp": 1707826531.111369}
{"name": "vdk.plugin.control_cli_plugin.execution_skip", "filename": "execution_skip.py", "lineno": 106, "funcName": "_skip_job_if_necessary", "message": "Job : hello-world, Team : my-team, Log config: LOCAL, execution_id: 7f829003-d1d9-4991-be9c-696e5ebca7f2-1707826531", "level": "INFO", "timestamp": 1707826531.1114}
{"name": "root", "filename": "execution_skip.py", "lineno": 111, "funcName": "_skip_job_if_necessary", "message": "Local execution, skipping parallel execution check.", "level": "INFO", "timestamp": 1707826531.111434}
{"name": "vdk.plugin.sqlite.sqlite_connection", "filename": "sqlite_connection.py", "lineno": 29, "funcName": "new_connection", "message": "Creating new connection against local file database located at: /var/folders/5d/bbd89v315777n_d24q2wznlh0000gr/T/vdk-sqlite.db", "level": "INFO", "timestamp": 1707826531.111862}
...
```

But it's also what vdk-core does today.

Reply:
If `vdk list` and `vdk deploy` implement the job hooks, their logs won't be affected if we choose to configure at `initialize_job`

Reply:
Yeah, you are correct, we almost certainly may want the option to format logs earlier in cloud (for vdk run) 

Do communicate that it might be good idea to set this for cloud with @dakodakov  or @mivanov1988 (though they may see the tag?)

Reply:
I set it to True by default. If they decide to use structlog in the cloud, we can have this discussion.